pyro/infer/enum.py

```diff
@@ -34,6 +34,8 @@ def get_importance_trace(graph_type, max_iarange_nesting, model, guide, *args, *
     Returns a single trace from the guide, and the model that is run
     against it.
     """
+    guide = poutine.broadcast(guide)
```

Comment:
With this change, can we remove `poutine.broadcast` annotation from our examples completely (since it will be done automatically behind the scenes)?

Reply:
Well, we still need `poutine.broadcast` for the model, and we'll still need it to run the guide under vanilla `Trace_ELBO`. I think the clearest policy would be to use `poutine.broadcast` for any model/guide that requires broadcasting when run outside of SVI.

Reply:
Yes, we'll still need it for HMC for instance. We are using `poutine.broadcast` on the model here too, and `Trace_ELBO` also calls `get_importance_trace`, so I think that should be covered..